src/Raven.Server/Documents/CollectionRunner.cs

```diff
@@ -85,36 +85,42 @@ public virtual Task<IOperationResult> ExecuteDelete(string collectionName, long
                     {
                         foreach (var document in GetDocuments(context, collectionName, startEtag, internalQueryOperationStart, OperationBatchSize, isAllDocs, DocumentFields.Id))
                         {
-                            internalQueryOperationStart++;
+                            using (document)
```

Comment:
We dispose here only the document id and we collect here `OperationBatchSize` which is 1024 IIRC.